Framework/MDAlgorithms/src/SmoothMD.cpp

```diff
@@ -161,10 +161,10 @@ const std::string SmoothMD::summary() const { return "Smooth an MDHistoWorkspace
  * @param weightingWS : Weighting workspace (optional)
  * @return Smoothed MDHistoWorkspace
  */
-IMDHistoWorkspace_sptr SmoothMD::hatSmooth(const IMDHistoWorkspace_const_sptr &toSmooth, const WidthVector &widthVector,
-                                           OptionalIMDHistoWorkspace_const_sptr weightingWS) {
+IMDHistoWorkspace_sptr SmoothMD::hatSmooth(IMDHistoWorkspace_const_sptr toSmooth, const WidthVector &widthVector,
+                                           IMDHistoWorkspace_sptr weightingWS) {
 
-  const bool useWeights = weightingWS.is_initialized();
+  const bool useWeights = weightingWS.get() != 0;
```

Comment:
I think we can just do
```cpp
const bool useWeights = (weightingWS != nullptr);
```
without needing to extract the pointer?

Reply:
Done :+1: